### PR TITLE
Add networking e2e tests

### PR DESCRIPTION
**Pull request by jinghanm**
## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/tests/e2etest/NetworkingTests.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            public string IpAddresses { get; set; }
+            public string SubnetMasks { get; set; }
+            public string DefaultGateways { get; set; }
+            public string DnsServers { get; set; }
+            public string DhcpEnabled { get; set; }
+            public string Enabled { get; set; }
+            public string Connected { get; set; }
+        }
+
+        public void RegexTest(string testString, string patternString)
```
Comment (hachotze) [inline src/tests/e2etest/NetworkingTests.cs:31]: Consider extending the function name to something like 'RegexTest_SingletonInterfaceProperties" for descriptiveness.

File: src/tests/e2etest/NetworkingTests.cs [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            string[] result = Regex.Split(testString, splitPattern);
+            foreach (string piece in result)
+            {
+                Assert.True(regexPattern.IsMatch(piece));
+            }
+        }
+
+        public void RegexTestCommaSplit(string testString, string piecePatternString, string smallPatternString)
+        {
+            string splitPattern = ";";
```
Comment (hachotze) [inline src/tests/e2etest/NetworkingTests.cs:44]: Consider changing 'splitPattern' and 'splitPatternComma' to something like 'interfaceDelimiter' and 'propertyDelimiter', respectively.

File: src/tests/e2etest/NetworkingTests.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+using System;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+using System.Collections.Generic;
+
+namespace e2etesting
+{
+    public class NetworkingTests : E2eTest
+    {
+        string ComponentName = "Networking";
```
Comment (AhmedBM) [inline src/tests/e2etest/NetworkingTests.cs:16]: should be `const`

File: src/tests/e2etest/NetworkingTests.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        string PropertyName = "NetworkConfiguration";
```
Comment (AhmedBM) [inline src/tests/e2etest/NetworkingTests.cs:17]: should be `const`

File: src/tests/e2etest/NetworkingTests.cs [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+                {
+                    Assert.True(smallRegexPattern.IsMatch(token));
+                }
+            }
+        }
+
+        [Test]
+        public void NetworkingTest_Get()
+        {
+            if ((GetTwin().ConnectionState == DeviceConnectionState.Disconnected) || (GetTwin().Status == DeviceStatus.Disabled))
```
Comment (AhmedBM) [inline src/tests/e2etest/NetworkingTests.cs:64]: first conditional argument should use `GetNewTwin()` in order to have the most up-to-date twin

File: src/tests/e2etest/NetworkingTests.cs [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+                Assert.Fail("Module is disconnected or is disabled");
+            }
+
+            var deserializedReportedObject = JsonSerializer.Deserialize<Networking>(GetTwin().Properties.Reported[ComponentName][PropertyName].ToString());
+
+            // Test InterfaceTypes
+            string patternString = @"([A-Za-z0-9]+)\=([A-Za-z0-9]+)";
+            RegexTest_SingletonInterfaceProperties(deserializedReportedObject.InterfaceTypes, patternString);
+
+            string piecePatternString = @"[A-Za-z0-9]+\=(.*)";
```
Comment (AhmedBM) [inline src/tests/e2etest/NetworkingTests.cs:75]: piece pattern? Not sure what that means, maybe a better name?

**Review by AhmedBM — CHANGES_REQUESTED** [review]
mostly minor comments, except i would recommend using `GetNewTwin` as the first call in order to ensure an up-to-date twin to work with.

File: src/tests/e2etest/NetworkingTests.cs (hunk 5 above)
Comment (AhmedBM) [reply src/tests/e2etest/NetworkingTests.cs:64]: actually i take this back, if this is a reported only module, i would just make use of `GetTwin`

File: src/tests/e2etest/NetworkingTests.cs (hunk 4 above)
Comment (jinghanm) [reply src/tests/e2etest/NetworkingTests.cs:17]: Fixed! Thanks Ahmed.

**Review by hachotze — DISMISSED** [review]

**Review by AhmedBM — COMMENTED** [review]
other than the unneeded `IsSameByJson`, it looks good

File: src/tests/e2etest/E2ETest.cs [hunk 7]
```diff
@@ -262,6 +262,13 @@ public static void AreEqualByJson(object expected, object actual)
             Assert.AreEqual(expectedJson, actualJson);
         }
 
+        public static bool IsSameByJson(object expected, object actual)
```
[the remark this answers is not in the document]
Comment (jinghanm) [reply src/tests/e2etest/E2ETest.cs]: Thanks Ahmed. IsSameByJson was used by settings tests, but is no longer needed. I have removed it.

**Review by MariusNi — DISMISSED** [review]
:clock1:

**Review by MariusNi — APPROVED** [review]
:shipit:

**Review by AhmedBM — DISMISSED** [review]

**Review by MariusNi — APPROVED** [review]
:shipit:

**Review by AhmedBM — APPROVED** [review]

**Pull request merged**